cluster-autoscaler/clusterstate/clusterstate.go

```diff
@@ -323,7 +323,7 @@ func (csr *ClusterStateRegistry) IsNodeGroupHealthy(nodeGroupName string) bool {
 	unjustifiedUnready := 0
 	// Too few nodes, something is missing. Below the expected node count.
 	if readiness.Ready < acceptable.MinNodes {
-		unjustifiedUnready += acceptable.MinNodes - readiness.Ready
+		unjustifiedUnready += acceptable.MinNodes - readiness.Ready - readiness.NotStarted
```

Comment:
It seems like with this change in common case (when CA was not restarted, and was originator of the change of size of MIG) we will account for upcoming nodes twice.
Once in updateAcceptableRanges when MinNodes is computed. And then here.

Reply:
Right. It's probably less harmful this way (at most, we'll consider the node group healthy a little while longer while waiting for scale up), but how about using max from {scaleUpRequests, notStarted} when computing minNodes?

https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/clusterstate/clusterstate.go#L482

Reply:
It would not change anything in common case, as scaleUpRequest would only disappear if all the requested nodes are up, right?